executing/executing.py

```diff
@@ -509,9 +530,556 @@ def visit_ClassDef(self, node):
+                opname (string|Seq[stirng]): inst.opname has to be equal or in `opname`
+                **args: every arg has to match inst.arg
+
+            Returns:
+                True if all conditions match the instruction
+
+            """
+            if isinstance(opname, tuple):
+                if instruction.opname not in opname:
+                    return False
+            else:
+                if instruction.opname != opname:
+                    return False
+            return all(getattr(instruction, k) == v for k, v in args.items())
+
+        def node_match(node_type, **args):
+            """
+            match the ast-node
+
+            Parameters:
+                node_type: type of the node
+                **args: every `arg` has to be equal `node.arg`
+                        or `node.arg` has to be an instance of `arg` if it is a type.
+            """
+            return isinstance(node, node_type) and all(
+                isinstance(getattr(node, k), v)
+                if isinstance(v, type)
+                else getattr(node, k) == v
+                for k, v in args.items()
+            )
+
+        def mangled_name(node, name=None):
+            """
+
+            Parameters:
+                node: the node which should be mangled
+                name: the name of the node
+
+            Returns:
+                The mangled name of `node`
+            """
+            if name is None:
+                if isinstance(node, ast.Attribute):
+                    name = node.attr
+                elif isinstance(node, ast.Name):
+                    name = node.id
+                else:
+                    name = node.name
+
+            if name.startswith("__") and not name.endswith("__"):
+                parent = node.parent
+                while True:
+                    if isinstance(parent, ast.ClassDef):
+                        return "_" + parent.name.lstrip("_") + name
+
+                    if not hasattr(parent, "parent"):
+                        break
+                    parent = parent.parent
+
+            return name
+
+        if op_name == "CACHE":
+            return
+
+        if inst_match("CALL") and node_match((ast.With, ast.AsyncWith)):
+            # call to context.__exit__
+            return
+
+        if inst_match(("CALL", "LOAD_FAST")) and node_match(
+            (ast.ListComp, ast.GeneratorExp, ast.SetComp, ast.DictComp)
+        ):
+            # call to the generator function
+            return
+
+        if inst_match(("CALL", "CALL_FUNCTION_EX")) and node_match(ast.ClassDef):
+            return
+
+        if inst_match(("COMPARE_OP", "IS_OP", "CONTAINS_OP")) and node_match(
+            ast.Compare
+        ):
+            return
+
+        if inst_match("LOAD_NAME", argval="__annotations__") and node_match(
+            ast.AnnAssign
+        ):
+            return
+
+        if (
+            (inst_match("LOAD_METHOD", argval="join") or inst_match("CALL"))
```

Comment:
```suggestion
            (inst_match("LOAD_METHOD", argval="join") or inst_match(("CALL", "BUILD_STRING")))
```

instead of the separate check below?

Reply:
might work. I will try it. 
This stuff was created case by case by running over python source code. I already unified some cases but there are probably more.